Resync `css/cssom` from WPT Upstream

Ahmad-S792 opened the pull request:
#### 554a7e52f922f2e8a00258fd5279f028d4bc35e8
<pre>
Resync `css/cssom` from WPT Upstream
<a href="https://bugs.webkit.org/show_bug.cgi?id=303591">https://bugs.webkit.org/show_bug.cgi?id=303591</a>
<a href="https://rdar.apple.com/165880975">rdar://165880975</a>

Reviewed by Brandon Stewart.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/bb493cfb216722d620698e9858496cd4c8090d0e">https://github.com/web-platform-tests/wpt/commit/bb493cfb216722d620698e9858496cd4c8090d0e</a>

* LayoutTests/imported/w3c/web-platform-tests/css/cssom/StyleSheetList-constructable-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/StyleSheetList-constructable-with-style-recalc-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/StyleSheetList-constructable-with-style-recalc.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/StyleSheetList-constructable.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/WEB_FEATURES.yml:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/caretRangeFromPoint-replace-document.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/caretRangeFromPoint-replace-document.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/caretRangeFromPoint-textarea-transform.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/caretRangeFromPoint-textarea-transform.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/caretRangeFromPoint.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/caretRangeFromPoint.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/cssstyledeclaration-csstext-all-shorthand.html:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/getComputedStyle-insets-relpos-inline.html:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/historical-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/historical.html:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/shorthand-values-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/shorthand-values.html:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/stylesheet-dom-mutation-event-crash.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/w3c-import.log:

Canonical link: <a href="https://commits.webkit.org/303990@main">https://commits.webkit.org/303990@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0b1fd9076ff7411ee422b0641f9416d82686c4cb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/134144 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/6655 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/45353 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/141724 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/86207 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/09f676f9-f9cb-4c07-addf-0ff90cf438d3) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/7191 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/6519 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/102582 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/69873 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/b5ed4d4d-cede-48e0-9f5c-2bcb8529c271) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/137091 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/5013 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/120240 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/83377 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/c6f91338-bc13-4a84-b253-77584d6cddd3) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/4893 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/2528 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/1539 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/114064 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/38371 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/144370 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/6325 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/38949 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/110957 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/6407 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/5327 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/111200 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28219 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/4759 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/116508 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/60095 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/6377 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/34731 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/6223 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/6468 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/6331 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

stwrt commented on LayoutTests/imported/w3c/web-platform-tests/css/cssom/caretRangeFromPoint-replace-document.tentative-expected.txt:
```diff
@@ -0,0 +1 @@
+
```
Is this expected to have nothing there?

Ahmad-S792 replied:
I was also confused and then noticed that it was adding 'textarea' and then replacing it. So it does not generate output. I could change to append but then it might not test what it intends to test but this test definitely need second look.

stwrt replied:
We can always import this PR. Then make any changes to the test, and finally export those changes.

stwrt reviewed (approved):
LGTM if tests are passing

Ahmad-S792 committed:
Resync `css/cssom` from WPT Upstream
https://bugs.webkit.org/show_bug.cgi?id=303591
rdar://165880975

Reviewed by Brandon Stewart.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/bb493cfb216722d620698e9858496cd4c8090d0e

* LayoutTests/imported/w3c/web-platform-tests/css/cssom/StyleSheetList-constructable-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/StyleSheetList-constructable-with-style-recalc-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/StyleSheetList-constructable-with-style-recalc.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/StyleSheetList-constructable.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/WEB_FEATURES.yml:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/caretRangeFromPoint-replace-document.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/caretRangeFromPoint-replace-document.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/caretRangeFromPoint-textarea-transform.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/caretRangeFromPoint-textarea-transform.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/caretRangeFromPoint.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/caretRangeFromPoint.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/cssstyledeclaration-csstext-all-shorthand.html:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/getComputedStyle-insets-relpos-inline.html:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/historical-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/historical.html:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/shorthand-values-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/shorthand-values.html:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/stylesheet-dom-mutation-event-crash.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/w3c-import.log:

Canonical link: https://commits.webkit.org/303990@main
